pkg/operation/botanist/etcdencryption.go

```diff
@@ -164,16 +260,36 @@ func (b *Botanist) RewriteShootSecretsIfEncryptionConfigurationChanged(ctx conte
 		return b.CheckSums[common.EtcdEncryptionSecretName]
 	}()
 
-	if err := b.rewriteShootSecretsIfChecksumChanged(ctx, checksum); err != nil {
+	err := b.rewriteShootSecretsIfChecksumChanged(ctx, checksum)
+	if err != nil {
 		return err
 	}
 
-	gardenSecret := &corev1.Secret{ObjectMeta: kutil.ObjectMetaFromKey(common.GardenEtcdEncryptionSecretKey(b.Shoot.Info.Namespace, b.Shoot.Info.Name))}
-	_, err := controllerutil.CreateOrUpdate(ctx, b.K8sGardenClient.Client(), gardenSecret, func() error {
-		kutil.SetMetaDataAnnotation(gardenSecret, common.EtcdEncryptionChecksumAnnotationName, checksum)
+	shootState := &gardencorev1alpha1.ShootState{ObjectMeta: kutil.ObjectMeta(b.Shoot.Info.Namespace, b.Shoot.Info.Name)}
+	_, err = controllerutil.CreateOrUpdate(ctx, b.K8sGardenClient.Client(), shootState, func() error {
```

Comment:
Again, a lot of duplicate code. Can't we build a helper function that takes the relevant information about the etcd encryption data and stores it in the shootstate?